src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
```

Comment:
I think if that's done, it should be clamped to a maximum. Otherwise `new []{1,2}.SkipLast(10000)` would create a 10000-element capacity queue to which it only adds two elements. (There are cases where we can be more confident in the count, in which case we should go ahead and use that count, though many of those would offer further optimisations akin to those currently in place for `Skip()` anyway).